### ISSUE TITLE
cordova prepare error

### ISSUE THREAD

**Issue by jinchan-ck** [RocketChat/Rocket.Chat.Cordova#21]
I followed the readme to setup Rocket.Chat.Cordova, but got this:

```
➜  Rocket.Chat.Cordova git:(develop) cordova prepare    
Error: Parameter 'url' must be a string, not undefined
```

I can not find any error definition like that from the project, anyone know what happened?

**Comment by jinchan-ck** [comment]
This is some of my env version:

```
➜  Rocket.Chat.Cordova git:(develop) ✗ cordova -v
5.4.0
➜  Rocket.Chat.Cordova git:(develop) ✗ node -v
v0.12.4

```

**Comment by graywolf336** [comment]
Try to run the other commands and not prepare, see if that works.

**Comment by GaWr26** [comment]
same error here. Get it with all the commands..

**Comment by jpmccu** [comment]
I get the same.

```
$ cordova -v 
5.4.0
$ node -v
v0.12.7
```

**Comment by jinchan-ck** [comment]
@graywolf336 `cordova emulate ios` got the same error:

```
➜  Rocket.Chat.Cordova git:(develop) ✗ cordova emulate ios
Error: Parameter 'url' must be a string, not undefined
```

**Comment by mitchrosu** [comment]
It seems to be connected with anything ios related. Running 'cordova platform update android' works fine, but running 'cordova platform update ios' returns the error "Error: Parameter 'url' must be a string, not undefined".

**Comment by graywolf336** [comment]
Ah okay, so that's why I didn't get the error as I'm on Windows and so I can only do android. I will try and look into this tonight when I get home if someone doesn't beat me to it.

**Comment by mitchrosu** [comment]
I've narrowed it down to the config.xml file. When I remove:

```
<allow-navigation origin="*" />
```

From line 27 the function loads correctly and doesn't return the error. Any ideas for what could be causing this line to break it? I'm not familiar with cordova so I'm going through the docs now to figure out what's wrong with it, but if you have any ideas let me know.

**Comment by mitchrosu** [comment]
Okay, straight from the docs:

```
<!-- A wildcard can be used to whitelist the entire network,
     over HTTP and HTTPS.
     *NOT RECOMMENDED* -->
<allow-navigation href="*" />

<!-- The above is equivalent to these three declarations -->
<allow-navigation href="http://*/*" />
<allow-navigation href="https://*/*" />
<allow-navigation href="data:*" />
```

So I replaced that line with the bottom 3 lines and it builds just fine without losing the intended functionality. Does this solution work for you or is there anything wrong with it @graywolf336 ?

**Comment by graywolf336** [comment]
Sounds good to me. How about this, submit a pull request and then we'll
test it and see if it does change anything. If it works, we'll merge the
pull request and go from there.

**Comment by geekgonecrazy** [comment]
:+1: for sure the best route

**Comment by rodrigok** [comment]
Closed via #23
